src/parcels/mount-parcel.js

```diff
@@ -90,38 +91,47 @@ export function mountParcel(config, customProps) {
       : NOT_BOOTSTRAPPED,
     customProps,
     parentName: toName(owningAppOrParcel),
+    internalUnmountPromise: undefined,
     unmountThisParcel() {
-      return mountPromise
-        .then(() => {
-          if (parcel.status !== MOUNTED) {
-            throw Error(
-              formatErrorMessage(
-                6,
-                __DEV__ &&
-                  `Cannot unmount parcel '${name}' -- it is in a ${parcel.status} status`,
-                name,
-                parcel.status
-              )
-            );
-          }
-          return toUnmountPromise(parcel, true);
-        })
-        .then((value) => {
-          if (parcel.parentName) {
-            delete owningAppOrParcel.parcels[parcel.id];
-          }
-
-          return value;
-        })
-        .then((value) => {
-          resolveUnmount(value);
-          return value;
-        })
-        .catch((err) => {
-          parcel.status = SKIP_BECAUSE_BROKEN;
-          rejectUnmount(err);
-          throw err;
-        });
+      // In some cases it is possible for a parcel to be in an unmounting status when `unmountThisParcel` is called a second time.
+      // https://github.com/single-spa/single-spa/issues/1184
+      if (parcel.status === UNMOUNTING && !!parcel.internalUnmountPromise) {
+        return parcel.internalUnmountPromise;
+      } else {
+        const unmountPromise = mountPromise
+          .then(() => {
+            if (parcel.status !== MOUNTED) {
+              throw Error(
+                formatErrorMessage(
+                  6,
+                  __DEV__ &&
+                    `Cannot unmount parcel '${name}' -- it is in a ${parcel.status} status`,
+                  name,
+                  parcel.status
+                )
+              );
+            }
+            return toUnmountPromise(parcel, true);
+          })
+          .then((value) => {
+            if (parcel.parentName) {
+              delete owningAppOrParcel.parcels[parcel.id];
+            }
+
+            return value;
+          })
+          .then((value) => {
+            resolveUnmount(value);
+            return value;
```

Comment:
Deleting the internalMountPromise might be helpful for garbage collection.

```suggestion
            delete this.internalMountPromise;
            return value;
```